app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,27 @@
+class PasswordResetsController < ApplicationController
+  def show
+    user = User.find_by(token: params[:id])
+    if user
+      @token = user.token
+    else
+      redirect_to expired_token_path
+    end
+    
+  end
+
+  def create
+    user = User.find_by(token: params[:token])
+    if user
+      user.password = params[:password]
+      user.generate_token
+      user.save
```

Comment:
This concept isn't good, What happens when the user inputs invalid password? or none? Also when you set password then you also save user instance. This should be safely done by `.update`.

Take a look here:

```ruby
def create
    user = User.find_by(token: params[:token])
    if user && user.update(password: params[:password])
      user.destroy_token
      flash[:notice] = "Password was reset"
      redirect_to sign_in_path
    else
      redirect_to expired_token_path
    end
  end
```

Also you do not want to generate another token after the user spends it. Its a huge threat, if someone possibly steals your data its like opened way to take the other account.